### PR TITLE
Upgrade to rc5, ngModules

### PR DESCRIPTION
**Pull request by nikolasleblanc**
Setup a giant `sample-app.module.ts` that could probably be broken into smaller modules, but for the time being, in the interest of getting to RC5 and adopting ngModules.

### CODE REVIEW

**Comment by nikolasleblanc** [comment]
Now I'm just fighting with the robot tests, but all unit tests are passing.

File: webpack/plugins.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const prodPlugins = [
   new webpack.optimize.OccurenceOrderPlugin(),
   new webpack.optimize.DedupePlugin(),
   new webpack.optimize.UglifyJsPlugin({
-    mangle: true,
+    mangle: { keep_fnames: true },
```
Comment (SethDavenport) [inline webpack/plugins.js]: This is broken again in RC5?  Huh.

**Comment by SethDavenport** [comment]
Looks good.

Going forward we could consider breaking it into a few smaller modules I guess, just to keep the boilerplace in index.ts down.

Maybe:
* navigation
* authentication
* counter
* core (redux stuff under store, actions and epics?)
* api

Just spitballing. Might be the kick in the pants we need to move everything above store to a more feature-based setup.

That can be a separate PR though.

File: webpack/plugins.js (hunk 1 above)
Comment (nikolasleblanc) [inline webpack/plugins.js]: Yup!

**Pull request merged**